### PR TITLE
Split out subclass and item category from loadouts

### PR DESCRIPTION
**Pull request by ryan-rushton**
Splitting out the item category and subclass components to be used in v2 of the loadout drawer.

No actual changes here, just pulling out some components and I also had to copy paste some scss due to weird competing priority of the composed vs override values.

## PR's
- https://github.com/DestinyItemManager/DIM/pull/7552 **(this one)**
- https://github.com/DestinyItemManager/DIM/pull/7555

### CODE REVIEW

File: src/app/loadout/Loadouts.m.scss [hunk 1]
```diff
@@ -210,11 +117,6 @@
   margin-top: 8px;
 }
 
-.subclassMods {
-  composes: modsGrid;
```
Comment (ryan-rushton) [inline src/app/loadout/Loadouts.m.scss:214]: This is the compose I called out not being able to use in the description.

Comment (ryan-rushton) [inline src/app/loadout/Loadouts.m.scss:214]: This is the compose I called out not being able to use in the description.

File: src/app/loadout/loadout-ui/Subclass.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { AppIcon, powerActionIcon } from 'app/shell/icons';
+import { getSocketsByIndexes } from 'app/utils/socket-utils';
+import { SocketCategoryHashes } from 'data/d2/generated-enums';
+import React, { useMemo } from 'react';
+import { createGetModRenderKey } from '../mod-utils';
+import PlugDef from './PlugDef';
+import styles from './Subclass.m.scss';
+
+/** The subclass section used in the loadouts page and drawer */
+export default function Subclass({
```
Comment (ryan-rushton) [inline src/app/loadout/loadout-ui/Subclass.tsx:17]: I would like a better name for this but had nothing really at the time. Best I could think of was `SubclassSection` but its not really descriptive.

Comment (bhollis) [reply src/app/loadout/loadout-ui/Subclass.tsx:17]: I'm generally down with longer names like `LoadoutSubclassSection`

**Review by bhollis — APPROVED** [review]

**Pull request merged**